pkgs/development/python-modules/cryptography/default.nix

```diff
@@ -50,8 +52,10 @@ buildPythonPackage rec {
     cffi
   ] ++ [
     rustPlatform.cargoSetupHook
```

Comment:
I would expect setuptooks to go into nativeBuildInputs because they are run on the build platform and change settings for the compiler but I might be wrong here.

Reply:
> I would expect setuptooks to go into nativeBuildInputs because they are run on the build platform

I think part of the problem is that `make-rust-platform.nix` isn't using `makeScopeWithSplicing`, so the splicing mechanism isn't working properly for it.  This is one of the reasons I encourage people to avoid using `rustPlatform` whenever possible (as @Cynerd pointed out, sometimes you can't).

Personally I don't understand why we need a `rustPlatform`; it's very confusing.  If you "flatten" `rustPlatform` and `rustPlatform.rust` you get only seven attributes.  Why not just put all seven of them in `all-packages.nix` and eliminate `rustPlatform`?

Nixpkgs has a *lot* of subtle bugs when it comes to dealing with things (like `rustPlatform`) in `all-packages.nix` that are attrsets-of-packages instead of just packages.  In a perfect world we would fix those bugs, but I'm not sure that is feasible on a reasonable timeline.

Reply:
It appears that you have applied this suggestion.